### DISCUSSION TITLE
[AX] Provide a Quicker New Element Workflow from Related Element Fields

### DISCUSSION THREAD

**Discussion by kylecotter** [craftcms/cms#11379] [Ideas] [answered]
I think it’d be a useful AX improvement to add a shortcut for creating a new entry by means of a secondary button or similar instead of having to trigger the modal first and then click “new entry” when using a related entry field. For times when you know you want to relate new content and not something already existing.

We’re using more and more a paradigm consisting of related entries and slide-overs vs. nested field approaches (e.g not super table within matrix etc) and this would streamline that content creation process for content editors where the workflow is - create a new “Page” entry and use a related Entries field for (primarily) _creating new_ “Component” entries, and selecting existing entries is a secondary added bonus.
![image (1)](https://user-images.githubusercontent.com/505894/171876905-1c3c3aa3-03e8-4adf-a7b9-7308d9b94c09.png)

**Comment by brandonkelly** [comment]
Agree that makes sense – similar to how Assets fields have an “Upload files” button beside the “Add an asset” button.

**Reply by brandonkelly** [reply]
I was hoping to get this into Entries fields for Craft 4.4, as it would pair nicely with the new “Maintain Hierarchy” setting (#11749), but ran into a snag: we need to come up with some way to ensure that it’s only possible to create entries that would be selectable via the “Selectable Entries Condition” setting.

**Reply by kylecotter** [reply]
I'd imagine that same scenario comes up in the traditional flow of "Add an entry" -> Modal -> "New entry"?

What if the entry were allowed to be created and added to the field, but creates a validation error on save of the parent indicating the related element doesn't meet the selection criteria?

e.g. similar to 
![SCR-20230222-k1a](https://user-images.githubusercontent.com/505894/220737400-7597acf7-3416-43f8-a484-7d50fa4019bb.png)

**Reply by brandonkelly** [reply]
> I'd imagine that same scenario comes up in the traditional flow of "Add an entry" -> Modal -> "New entry"?

No, because those entries aren’t automatically added. If the resulting entry doesn’t match the selection criteria, it won’t show up in the modal window te begin with.

**Reply by kylecotter** [reply]
Ah, gotcha. I’d argue that could be a double-edged AX flow too.

If a Content editor creates entry via modal -> forgets to set one field that makes that entry match the selection criteria -> has no way of finding that again without going out to the main Entries page.

**Reply by brandonkelly** [reply]
Agree, it’s not perfect. Ideally we come up with a better solution for both.

**Reply by thupsi** [reply]
A reasonable compromise that I can think of is that the `New element` button would only be an option when creating a relations field that has no conditions OR has conditions that do not pose a problem. 

For example if only relations from certain sections are allowed, then the `New element` button could be a dropdown button like in the entry index pages (new entrytype1, new entrytype2 etc).

That way relation fields that don't have complex conditions would still benefit from the faster workflow.

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.8.0 is out with inline search for relation fields, and inline entry creation for Entries fields (#17497) 🎉
